src/passkey_child/passkey_child_devices.c

```diff
@@ -244,3 +244,25 @@ get_device_options(fido_dev_t *dev, struct passkey_data *_data)
 
     return ret;
 }
+
+errno_t
+get_device_pin_retries(fido_dev_t *dev, struct passkey_data *data,
+                       int *_pin_retries)
+{
+    int ret = EOK;
+
+    if (data->user_verification == FIDO_OPT_TRUE) {
+        ret = fido_dev_get_retry_count(dev, _pin_retries);
+        if (ret != FIDO_OK) {
+            DEBUG(SSSDBG_OP_FAILURE,
+                "fido_dev_get_retry_count failed [%d]: %s.\n",
+                ret, fido_strerr(ret));
+            goto done;
+        }
+    } else {
+        *_pin_retries = MAX_PIN_RETRIES;
```

Comment:
If `user verification == false` will GDM hide this value? I see that the `fido2-token` command returns `undefined` which I think is more indicative that the device has no PIN set but I guess GDM will expect an integer?

~~~
justin@fedora:~$ fido2-token -I /dev/hidraw5 | grep 'pin retri'
pin retries: undefined
~~~

Reply:
Right, GDM is expecting an integer and that's why I'm setting the highest possible value. GDM will only show a warning message when the number of attempts is below a threshold (probably 3), if the value is higher than that nothing will be shown to the user.